wicket-core/src/main/java/org/apache/wicket/markup/html/form/upload/FilesSelectedBehavior.js

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+;(function (undefined) {
+
+    'use strict';
+
+    if (typeof(Wicket.FilesSelected) === 'object') {
+        return;
+    }
+
+    /**
+     * Contains JS code for FilesSelectedBehavior.
+     */
+    Wicket.FilesSelected = {
+
+        // the id of temporary object holding files.
+        varId: function (componentId) {
+            return componentId + "_files";
+        },
+
+        /**
+         * Precondition to trigger (or not) server round-trip.
+         *
+         * @param inputField the file input field
+         * @returns {boolean} true if some files (file) were (was) selected,
+         */
+        precondition : function (inputField) {
+            var id = this.varId(inputField.id);
+            console.log(id)
```

Comment:
debug leftover

Reply:
Do we have a Wicket.log? On our private project I have added a log function what can be enabled via some flag you can change at browser console (to switch on debugging).

Reply:
Yes - https://github.com/apache/wicket/blob/048e5681df960018722237ee6254273e839b5fd2/wicket-core/src/main/java/org/apache/wicket/ajax/res/js/wicket-ajax-jquery.js#L189-L222
It could be enabled/disabled via DebugSettings#ajaxDebugModeEnabled

Reply:
I see... Thanks for info. I see you can maybe also type Wicket.Log.enable=true; in console to enable it for some page.

Reply:
Yes! It will survive until the next page reload.